pmd-dist/src/main/resources/scripts/pmd.bat

```diff
@@ -1,6 +1,54 @@
 @echo off
 set TOPDIR="%~dp0.."
 set OPTS=
-set MAIN_CLASS=net.sourceforge.pmd.PMD
+set COMMAND=%1
+set MAIN_CLASS=net.sourceforge.pmd.cli.PmdCli
 
-java %PMD_JAVA_OPTS% -classpath %TOPDIR%\conf;%TOPDIR%\lib\* %OPTS% %MAIN_CLASS% %*
+
+:: sets the jver variable to the java version, eg 901 for 9.0.1+x or 180 for 1.8.0_171-b11
+:: sets the jvendor variable to either java (oracle) or openjdk
+for /f tokens^=1^,3^,4^,5^ delims^=.-_+^"^  %%j in ('java -version 2^>^&1 ^| find "version"') do (
+  set jvendor=%%j
+  if %%l EQU ea (
+    set /A "jver=%%k00"
+  ) else (
+    set /A jver=%%k%%l%%m
+  )
+)
+
+Set "jreopts="
+:: oracle java 9 and 10 has javafx included as a module
+if /I "%jvendor%" EQU "java" (
+    if %jver% GEQ 900 (
+        if %jver% LSS 1100 (
+            :: enable reflection
+            Set jreopts=--add-opens javafx.controls/javafx.scene.control.skin=ALL-UNNAMED
```

Comment:
when going over this, I noticed that run shell script includes a bunch of other options that are missing here:

```
# The rest here is for RichtextFX
      options+=" --add-opens javafx.graphics/javafx.scene.text=ALL-UNNAMED"
      options+=" --add-opens javafx.graphics/com.sun.javafx.scene.text=ALL-UNNAMED"
      options+=" --add-opens javafx.graphics/com.sun.javafx.text=ALL-UNNAMED"
      options+=" --add-opens javafx.graphics/com.sun.javafx.geom=ALL-UNNAMED"
      # Warn of remaining illegal accesses
      options+=" --illegal-access=warn"
```

I don't know why, or how to do concat to a var in bat files.

Reply:
This seems to do the trick:

```suggestion
            SETLOCAL EnableDelayedExpansion
            rem java9 and java10 from oracle contain javafx as a module
            rem open internal module of javafx to reflection (for our TreeViewWrapper)
            set "jreopts=--add-opens javafx.controls/javafx.scene.control.skin=ALL-UNNAMED"
            rem The rest here is for RichtextFX
            set "jreopts=!jreopts! --add-opens javafx.graphics/javafx.scene.text=ALL-UNNAMED"
            set "jreopts=!jreopts! --add-opens javafx.graphics/com.sun.javafx.scene.text=ALL-UNNAMED"
            set "jreopts=!jreopts! --add-opens javafx.graphics/com.sun.javafx.text=ALL-UNNAMED"
            set "jreopts=!jreopts! --add-opens javafx.graphics/com.sun.javafx.geom=ALL-UNNAMED"
            rem Warn of remaining illegal accesses
            set "jreopts=!jreopts! --illegal-access=warn"

```

We are in parenthesis and the CMD shell will read and expand the vars only once - so a simple set "jreopts=%jreopts another option" doesn't work.
We need to enable EnableDelayedExpansion and use "!"...
See https://ss64.com/nt/delayedexpansion.html